source/addonStore/dataManager.py

```diff
@@ -162,6 +169,49 @@ def _cacheCompatibleAddons(self, addonData: str, cacheHash: Optional[str]):
 		with open(self._cacheCompatibleFile, "w", encoding="utf-8") as cacheFile:
 			json.dump(cacheData, cacheFile, ensure_ascii=False)
 
+	def _shouldCacheCompatibleAddonsBackup(self) -> bool:
+		if not os.path.exists(self._cacheCompatibleOldFile):
+			return True
+		resetNewAddons = config.conf["addonStore"]["resetNewAddons"]
+		if resetNewAddons == "startup":
+			return True
+		lastBackupTime = os.path.getmtime(self._cacheCompatibleOldFile)
+		lastBackupDate = datetime.fromtimestamp(lastBackupTime)
+		nowDate = datetime.now()
+		diffDate = nowDate - lastBackupDate
+		if resetNewAddons == "weekly" and diffDate.days >= 7:
+			return True
+		if resetNewAddons == "monthly" and diffDate.days >= 30:
+			return True
+		return False
+
+	def _getResetNewAddonsDate(self) -> str:
+		resetNewAddons = config.conf["addonStore"]["resetNewAddons"]
+		if resetNewAddons == "startup":
+			# Translators: Message presented in the new add-ons tab, informing that new add-ons will be reset at startup
+			return _("Will be reset at startup")
+		lastBackupTime = os.path.getmtime(self._cacheCompatibleOldFile)
+		lastBackupDate = datetime.fromtimestamp(lastBackupTime)
+		formattedLastBackupDate = lastBackupDate.strftime("%d/%m/%Y")
+		if resetNewAddons == "monthly":
+			timedeltaDays = 30
+		else:  # weekly
+			timedeltaDays = 7
+		nextResetDate = lastBackupDate + timedelta(days=timedeltaDays)
```

Comment:
The way this is presenting in the UI, makes it look like it's doing the math inversely. That is, on my system, if I don't choose all, the filtering date appears to be trying to show add-ons that are new between July 4th and August 3rd. I think it should be instead between June 3rd to July 4th.

Reply:
Hi @XLTechie , thanks for your review. No, add-ons will be considered new from 4 July, and will be reset on August. Add-on metadata doesn't have a timestamp, so NVDA will start saving all compatible add-ons cnsidering them as old add-ons, and add-ons published from 4 Jul will be considered as new add-ons. Hope this helps.

Reply:
@seanbudd how hard would it be to start embedding a generation timestamp in the JSON? I should think, not very.